### PR TITLE
DEV-5249 | exclude contributions with dummy pm id from portal

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

This solves for a bug whereby portal does not load for contributors who have one or more contributions with a provider payment method id value of "pm_this_is_temp_for_2445".  When contributors access the portal, the endpoint pulls in payment method data from Stripe, and there's an uncaught error around this ID because it's fake.

There's another ticket — [DEV-4967](https://news-revenue-hub.atlassian.net/browse/DEV-4967) — which will solve for the underlying problem (getting correct payment method ID on these contributions). This ticket just prevents that data state from breaking portal.

#### Why are we doing this? How does it help us?

Unbreaks portal for about 6k users

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

- [DEV-5249](https://news-revenue-hub.atlassian.net/browse/DEV-5249)
- [DEV-4967](https://news-revenue-hub.atlassian.net/browse/DEV-4967)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

[DEV-4693]: https://news-revenue-hub.atlassian.net/browse/DEV-4693?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEV-5249]: https://news-revenue-hub.atlassian.net/browse/DEV-5249?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DEV-4967]: https://news-revenue-hub.atlassian.net/browse/DEV-4967?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/contributions/tests/test_models.py [hunk 1]
```diff
@@ -2121,6 +2121,12 @@ def test_get_via_reversion_comment(self):
             reversion.set_comment(msg := "foo")
         assert set(Contribution.objects.get_via_reversion_comment(msg)) == {expected_contribution}
 
+    def test_exclude_exclude_dummy_payment_method_id(self, settings):
```
Comment (x110dc) [inline apps/contributions/tests/test_models.py:2124]: only one `exclude`?

File: revengine/settings/base.py [hunk 2]
```diff
@@ -669,3 +669,8 @@ class CurrencyDict(TypedDict):
 RETRIEVED_STRIPE_ENTITY_CACHE_TTL = 60 * 3
 
 SWITCHBOARD_ACCOUNT_EMAIL = os.getenv("SWITCHBOARD_ACCOUNT_EMAIL", None)
+
+# TODO @bw: Remove this after DEV-4693 is done
+# DEV-4693
```
Comment (x110dc) [inline revengine/settings/base.py:674]: DEV-4693 is marked as done already. Is that a typo and a different ticket should appear here?

**Review by x110dc — CHANGES_REQUESTED** [review]

File: apps/contributions/tests/test_models.py (hunk 1 above)
Comment (benjaminEwhite) [reply apps/contributions/tests/test_models.py]: 973b446b67

File: revengine/settings/base.py (hunk 2 above)
Comment (benjaminEwhite) [reply revengine/settings/base.py]: This was a typo (meant DEV-4963) and on top of that, had the ticket wrong.  Should be DEV-4967.

bab721dcb

**Review by x110dc — APPROVED** [review]

**Pull request merged**